django/db/backends/base/creation.py

```diff
@@ -183,13 +183,50 @@ def _create_test_db(self, verbosity, autoclobber, keepdb=False):
 
         return test_database_name
 
-    def destroy_test_db(self, old_database_name, verbosity=1, keepdb=False):
+    def clone_test_db(self, number, verbosity=1, autoclobber=False, keepdb=False):
+        """
+        Clone a test database.
+        """
+        source_database_name = self.connection.settings_dict['NAME']
+        target_database_name = self._get_test_db_clone_name(source_database_name, number)
+
+        if verbosity >= 1:
+            test_db_repr = ''
+            action = 'Cloning test database'
+            if verbosity >= 2:
+                test_db_repr = " ('%s')" % source_database_name
+            if keepdb:
+                action = 'Using existing clone'
+            print("%s for alias '%s'%s..." % (
+                action, self.connection.alias, test_db_repr))
+
+        # We could skip this call if keepdb is True, but we instead
+        # give it the keepdb param. See create_test_db for details.
+        self._clone_test_db(source_database_name, target_database_name, verbosity, keepdb)
```

Comment:
This is wrong for Oracle because the databases are not identified by `NAME` -- on Oracle, they're identified by `USER`. To make it work for both (and any other crazy scheme), `clone_test_db()` should be passing the whole settings dict down to `_get_test_db_clone_name()` and `_clone_test_db()`; otherwise, we need to override `clone_test_db()` itself.

See PR #4770